src/pip/_internal/cli/cmdoptions.py

```diff
@@ -99,6 +101,122 @@ def check_dist_restriction(options: Values, check_target: bool = False) -> None:
+        """
+        determine if a platform suffix has proper content and structure
+        for details about
+        """
+        # manylinux1, manylinux2010, manylinux2014, and linux should be followed
+        # by _<architecture>
+        linux_re = re.compile(r"(?P<arch>.+)")
+        match = linux_re.fullmatch(suffix)
+        if not match:
+            return False
+
+        arch = match.group("arch")
+        if prefix == "linux":
+            return arch in {"i386", "x86_64"}
+        if prefix in {"manylinux1", "manylinux2010"}:
+            return arch in {"i686", "x86_64"}
+        if prefix == "manylinux2014":
+            return arch in {
+                "aarch64",
+                "armv71",
+                "i686",
+                "ppc64",
+                "ppc64le",
+                "s390x",
+                "x86_64",
+            }
+        return False
+
+    def is_glibc_linux_arch(suffix: str) -> bool:
+        # manylinux alone should be followed by a _<major>_<minor>_<arch>
+        glibc_manylinux_re = re.compile(r"(?P<major>\d+)_(?P<minor>\d+)_(?P<arch>.+)")
+        glx = glibc_manylinux_re.fullmatch(suffix)
+        # PEP 600 recommends any tag of the form "manylinux_[0-9]+_[0-9]+_(.*)"
+        # be accepted by package indexes. Therefore, as long as the glibc
+        # major and minor versions are digits and there exists a value for
+        # the architecture, we should accept it. If they do not exist
+        # return False as improper input
+        return bool(glx)
+
+    def is_win_arch(suffix: str) -> bool:
+        # we don't need to check for a win32 given the calling function
+        # will enter here if this was a match. Otherwise, we do need
+        # to check if the architecture matches amd64.
+        win_re = re.compile(r"(?P<arch>)")
+        match = win_re.fullmatch(suffix)
+        return match.group("arch") == "amd64" if match else False
+
+    if not platform_options:
+        return
+
+    current_platform, *_ = platform.platform().lower().partition("-")
+    current_machine = platform.machine()
+    invalid_platforms = []
+    for platform_tag in platform_options:
+        platform_prefix, _, platform_suffix = platform_tag.partition("_")
+        if platform_prefix == "macosx":
+            if not is_macos_arch(platform_suffix):
+                invalid_platforms.append(platform_tag)
+        elif platform_prefix in {
+            "linux",
+            "manylinux1",
+            "manylinux2010",
+            "manylinux2014",
+        }:
+            if not is_linux_arch(platform_prefix, platform_suffix):
+                invalid_platforms.append(platform_tag)
+        elif platform_prefix == "manylinux":
+            if not is_glibc_linux_arch(platform_suffix):
+                invalid_platforms.append(platform_tag)
+        elif platform_prefix in ["win", "win32"]:
+            if not is_win_arch(platform_suffix):
+                invalid_platforms.append(platform_tag)
```

Comment:
I find it unnested easier to read.

```suggestion
        if platform_prefix == "macosx" and not is_macos_arch(platform_suffix):
            invalid_platforms.append(platform_tag)
        elif platform_prefix in {
            "linux",
            "manylinux1",
            "manylinux2010",
            "manylinux2014",
        } and not is_linux_arch(platform_prefix, platform_suffix):
            invalid_platforms.append(platform_tag)
        elif platform_prefix == "manylinux" and not is_glibc_linux_arch(platform_suffix):
            invalid_platforms.append(platform_tag)
        elif platform_prefix in ["win", "win32"] and not is_win_arch(platform_suffix):
            invalid_platforms.append(platform_tag)
```